### PR TITLE
chore: Activity screen refactoring & code decoupling

### PR DESCRIPTION
**Pull request by olerichter00**
Follows up on https://github.com/artsy/eigen/pull/10988

### Description

This is an attempt to refactor and decouple the Activity screen code to avoid bugs like the one fixed in https://github.com/artsy/eigen/pull/10988. I noticed that the artwork list component, which is used for many activity types, contains some type-specific elements and optimizations and tried to move them out of the component.

| Before | After (iOS) | After (Android) |
| --- | --- | --- |
|Simulator Screenshot - iPhone 15 Plus - 2024-11-04 at 12 33 48 | Simulator Screenshot - iPhone 15 Plus - 2024-11-04 at 12 34 25|Screenshot_20241104_130202 |
| --- | --- | --- |
|Simulator Screenshot - iPhone 15 Plus - 2024-11-04 at 12 35 05 |Simulator Screenshot - iPhone 15 Plus - 2024-11-04 at 12 35 11 |Screenshot_20241104_130334 |
| --- | --- | --- |
|Simulator Screenshot - iPhone 15 Plus - 2024-11-04 at 12 35 48 |Simulator Screenshot - iPhone 15 Plus - 2024-11-04 at 12 35 28 |Screenshot_20241104_130409 |
| --- | --- | --- |
|Simulator Screenshot - iPhone 15 Plus - 2024-11-04 at 12 36 07 |Simulator Screenshot - iPhone 15 Plus - 2024-11-04 at 12 36 26 | Screenshot_20241104_130527|

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-  

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Activity item screen code refactoring & decoupling (test setup: https://www.notion.so/artsy/Activity-Panel-cd0a9668895848a484b51bebb9723a8f?pvs=4#1faeb480c6d3432fa866718aa446ef5d) - ole

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Components/ArtworkGrids/MasonryArtworkGridItem.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { ContextModule, ScreenOwnerType } from "@artsy/cohesion"
 import { Flex, useScreenDimensions, useSpace } from "@artsy/palette-mobile"
 import { ArtworkGridItem_artwork$data } from "__generated__/ArtworkGridItem_artwork.graphql"
 import ArtworkGridItem, { PriceOfferMessage } from "app/Components/ArtworkGrids/ArtworkGridItem"
-import { PartnerOffer } from "app/Scenes/Activity/components/NotificationArtworkList"
+import { PartnerOffer } from "app/Scenes/Activity/components/PartnerOfferCreatedNotification"
```
Comment (damassi) [inline src/app/Components/ArtworkGrids/MasonryArtworkGridItem.tsx:5]: 👌

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Activity item screen code refactoring & decoupling (test setup: https://www.notion.so/artsy/Activity-Panel-cd0a9668895848a484b51bebb9723a8f?pvs=4#1faeb480c6d3432fa866718aa446ef5d) - ole - olerichter00)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6c6e0f7c06c5ba8c4b2c2aa4114f25970c2c7264
</p>

**Review by dariakoko — APPROVED** [review]

**Review by MrSltun — COMMENTED** [review]
This looks good!

But I can see that there are changes in the rendered elements code, maybe we should test it and take screenshots before and after to make sure that this changes doesn't introduce a visual bug?

**Comment by olerichter00** [comment]
> But I can see that there are changes in the rendered elements code, maybe we should test it and take screenshots before and after to make sure that this changes doesn't introduce a visual bug?

That's a very good idea 👍 I've tested the changes with all notification types but haven't taken screenshots. I'll add them to the PR.

**Pull request merged**